### PR TITLE
created vpc with reusable code using tfvars and variables

### PR DESCRIPTION
**Pull request by mamun1456**

### CODE REVIEW

File: Day-2/provider.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      source  = "hashicorp/aws"
+      version = ">=5.5.0"
+    }
+  }
+}
+
+# provider block
+provider "aws" {
+  region = var.region
+}
```
Comment (geoffrey1330) [inline Day-2/provider.tf]: could you please, add a newline at the end of file.

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

File: Day-2/provider.tf (hunk 1 above)
Comment (mamun1456) [reply Day-2/provider.tf]: added new lines at the end

File: Day-2/vpc.tf [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  subnet_id      = aws_subnet.public[count.index].id
+  route_table_id = aws_route_table.public.id
+}
+
+resource "aws_route_table_association" "private" {
+  count          = length(var.private_subnet_cidr)
+  subnet_id      = aws_subnet.private[count.index].id
+  route_table_id = aws_route_table.private[count.index].id
+}
+
```
Comment (geoffrey1330) [inline Day-2/vpc.tf:104]: could you please, reduce the end line to one also.

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

File: Day-2/variables.tf [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+variable "env" {}
+
+variable "instance_type" {}
+
+variable "vpc_cidr" {}
+
+variable "public_subnet_cidr" {}
+
+variable "private_subnet_cidr" {}
+
```
Comment (geoffrey1330) [inline Day-2/variables.tf:14]: could you please, reduce the end line to one also.

File: Day-2/terraform.tfvars [hunk 4]
```diff
@@ -0,0 +1,9 @@
+region              = "us-east-1"
+availability_zones  = ["us-east-1a", "us-east-1b", "us-east-1c"]
+env                 = "QA"
+instance_type       = "t2.micro"
+vpc_cidr            = "10.0.0.0/16"
+public_subnet_cidr  = ["10.0.0.0/24", "10.0.1.0/24", "10.0.2.0/24"]
+private_subnet_cidr = ["10.0.100.0/24", "10.0.101.0/24", "10.0.102.0/24"]
+
```
Comment (geoffrey1330) [inline Day-2/terraform.tfvars:8]: could you please, reduce the end line to one also.

File: Day-2/vpc.tf (hunk 2 above)
Comment (geoffrey1330) [inline Day-2/vpc.tf:104]: could you please, reduce the end line to one also.

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

**Review by geoffrey1330 — APPROVED** [review]

**Pull request merged**